src/ucp/proto/proto_common.inl

```diff
@@ -86,8 +86,8 @@ ucp_proto_request_zcopy_clean(ucp_request_t *req, unsigned dt_mask)
 static UCS_F_ALWAYS_INLINE void
 ucp_proto_request_zcopy_complete(ucp_request_t *req, ucs_status_t status)
 {
-    ucp_proto_request_zcopy_clean(req, UCP_DT_MASK_CONTIG_IOV);
-    ucp_datatype_iter_cleanup(&req->send.state.dt_iter, UCP_DT_MASK_CONTIG_IOV);
+    ucp_datatype_iter_cleanup(&req->send.state.dt_iter, 1,
+                              UCP_DT_MASK_CONTIG_IOV);
```

Comment:
`ucp_proto_request_zcopy_clean` includes `req->flags &= ~UCP_REQUEST_FLAG_PROTO_INITIALIZED;` and memory deregestration. Now memory deregestritaion is a part of cleanup. Isn't this an issue, that we no longer remove the flag in this method?

Reply:
it should not be an issue because ucp_proto_request_zcopy_complete() would release the request anyway